opendevin/controller/agent_controller.py

```diff
@@ -376,83 +385,32 @@ def set_initial_state(
         else:
             self.state = state
 
+        # when restored from a previous session, the State object will have history, start_id, and end_id
+        # connect it to the event stream
+        self.state.history.set_event_stream(self.event_stream)
+
+        # if start_id was not set in State, we're starting fresh, at the top of the stream
+        start_id = self.state.start_id
+        if start_id == -1:
+            start_id = self.event_stream.get_latest_event_id() + 1
```

Comment:
It will include 0, because `get_latest_event_id()` is `return self._cur_id - 1`, with a `cur_id` initially 0. Just re-tested... yes it does look a little suspicious 😅

Reply:
I should've looked at that closer, now the -1 makes much more sense to me 😂